externals/kyuubi-flink-sql-engine/src/test/scala/org/apache/kyuubi/engine/flink/operation/FlinkOperationSuite.scala

```diff
@@ -1282,6 +1282,26 @@ abstract class FlinkOperationSuite extends HiveJDBCTestHelper with WithFlinkTest
     assert(exception.getMessage === "Futures timed out after [60000 milliseconds]")
   }
 
+  test("test result fetch timeout but already fetched some data") {
+    withSessionConf()(Map(
+      ENGINE_FLINK_MAX_ROWS.key -> "10",
+      ENGINE_FLINK_FETCH_TIMEOUT.key -> "PT5S"))(Map.empty) {
+      withJdbcStatement() { statement =>
+        statement.execute("create table tbl_src_a (a bigint) with (" +
+          "'connector' = 'datagen', 'rows-per-second' = '1','number-of-rows' = '1')")
+        statement.execute("create table tbl_src_b (a bigint) with (" +
+          "'connector' = 'datagen', 'rows-per-second' = '0')")
+        val resultSet = statement.executeQuery(s"select a from tbl_src_a " +
+          s"union all select a from tbl_src_b")
+        var rows = 0
+        while (resultSet.next()) {
+          rows += 1
+        }
+        assert(rows === 1)
```

Comment:
Before the PR, the rows would be 0, right?

Reply:
Futures timed out after [5000 milliseconds]